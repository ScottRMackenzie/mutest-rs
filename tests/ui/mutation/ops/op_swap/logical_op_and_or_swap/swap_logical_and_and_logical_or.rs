//@ print-mutants
//@ build
//@ stdout
//@ stderr: empty
//@ mutation-operators: logical_op_and_or_swap

fn f() {
    let _ = (1_u32 as usize != 0) && (5_usize == 1);
    let _ = true && false && false && 1.0_f32 == 1 as f32;

    let _ = (99 != 0) || (13_usize == 1);
    let _ = true || false || false;

    let not_a_bool = 5 + 7 - 3;

    let _ = (not_a_bool + 1) == 10 || not_a_bool * 0 >= 1 && (true || false);

    if (false && true) as u32 > 0 {}

    match (33 > 1) || (2 == 2) {
        _ if (12 == 1 || not_a_bool > 0) as i8 > 0 => {}
        _ => {}
    }

    for _ in 0..((false || true) as usize) {}
}

#[test]
fn test() {
    f();
}
