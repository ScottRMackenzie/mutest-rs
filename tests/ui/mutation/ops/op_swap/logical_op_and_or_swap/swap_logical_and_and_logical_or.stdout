1 mutation
  - [logical_op_and_or_swap] swap logical operator `&&` for `||` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:8:13: 8:52
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `&&` for `||` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:9:13: 9:58
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `&&` for `||` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:9:13: 9:35
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `&&` for `||` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:9:13: 9:26
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `||` for `&&` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:11:13: 11:41
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `||` for `&&` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:12:13: 12:35
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `||` for `&&` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:12:13: 12:26
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `||` for `&&` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:16:13: 16:77
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `&&` for `||` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:16:39: 16:77
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `||` for `&&` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:16:63: 16:76
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `&&` for `||` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:18:9: 18:22
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `||` for `&&` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:20:11: 20:31
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `||` for `&&` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:21:15: 21:40
    <-(0)- test

1 mutation
  - [logical_op_and_or_swap] swap logical operator `||` for `&&` in f at tests/ui/mutation/ops/op_swap/logical_op_and_or_swap/swap_logical_and_and_logical_or.rs:25:19: 25:32
    <-(0)- test

14 mutants; 14 mutations; 14 safe; 0 unsafe (0 tainted); 0 batched; 14 unbatched
